server/src/main/java/io/druid/server/http/RulesResource.java

```diff
@@ -114,4 +114,16 @@ public Response getDatasourceRuleHistory(
                    .build();
   }
 
+  @GET
+  @Path("/history")
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response getDatasourceRuleHistory(
+      @QueryParam("interval") final String interval
+  )
+  {
+    Interval theInterval = interval == null ? null : new Interval(interval);
```

Comment:
If an invalid interval is specified it will throw illegalArgumentException to the user. 
Do you mean to throw any other exception to the user?